src/main/java/fi/aalto/cs/apluscourses/intellij/toolwindows/ExercisesToolWindowFactory.java

```diff
@@ -0,0 +1,14 @@
+package fi.aalto.cs.apluscourses.intellij.toolwindows;
+
+import com.intellij.openapi.project.Project;
+import fi.aalto.cs.apluscourses.ui.ExerciseList;
+import javax.swing.JComponent;
+import org.jetbrains.annotations.NotNull;
+
+public class ExercisesToolWindowFactory extends BaseToolWindowFactory {
+
+  @Override
+  protected JComponent createToolWindowContentInternal(@NotNull Project project) {
+    return new ExerciseList().getBasePanel();
```

Comment:
this is slightly smaller than the `ModulesToolWindowFactory` because it is undone?

Reply:
Yes (if I got your point right). We don't have Exercises tool window yet (just the placeholder) so this method just renders it. There are no viewmodel bindings, toolbar setups and such things here.